code/game/objects/items/implants/implant_abductor.dm

```diff
@@ -13,22 +13,46 @@
 		to_chat(imp_in, span_warning("You must wait [timeleft(on_cooldown)*0.1] seconds to use [src] again!"))
 		return
 
-	home.Retrieve(imp_in,1)
+	if(!home)
+		if(!link_pad())
+			imp_in.balloon_alert(imp_in, "no teleport pads detected!")
+			return
+
+	home.Retrieve(imp_in)
 	on_cooldown = addtimer(VARSET_CALLBACK(src, on_cooldown, null), cooldown , TIMER_STOPPABLE)
 
 /obj/item/implant/abductor/implant(mob/living/target, mob/user, silent = FALSE, force = FALSE)
 	if(..())
-		var/obj/machinery/abductor/console/console
-		if(ishuman(target))
-			var/datum/antagonist/abductor/A = target.mind.has_antag_datum(/datum/antagonist/abductor)
-			if(A)
-				console = get_abductor_console(A.team.team_number)
-				home = console.pad
-
-		if(!home)
-			var/list/consoles = list()
-			for(var/obj/machinery/abductor/console/C in GLOB.machines)
-				consoles += C
-			console = pick(consoles)
+		link_pad()
+		return TRUE
+
+/**
+ * Manages the process of linking a recall implant to an abductor pad
+ *
+ * Attempts to link the abductor implant to an abductor console. First, it tries to do so through the abductor's antag datum
+ * If not, a random teleport pad will be defaulted to. Returns TRUE if a home is found, and FALSE is one somehow is not.
+ */
+
+/obj/item/implant/abductor/proc/link_pad()
+	var/obj/machinery/abductor/console/console
+	if(ishuman(imp_in))
+		var/datum/antagonist/abductor/A = imp_in.mind.has_antag_datum(/datum/antagonist/abductor)
+		if(A)
+			console = get_abductor_console(A.team.team_number)
+			if(!console)
+				WARNING("Attempted to link [name] within [imp_in] to a pad using their abductor antagonist datum, however no associated machinery exists for their team.")
+				return FALSE
 			home = console.pad
+
+	if(!home) //If we still cannot find a home associated with our team, we just pick a random pad and make it our own.
+		var/list/consoles = list()
+		for(var/obj/machinery/abductor/console/C in GLOB.machines)
+			consoles += C
```

Comment:
single letter cleanup please. preferably not console again, maybe `potential_console`? unsure